### PR TITLE
Update the NOC root certificate design based on discussion feedback

### PR DESCRIPTION
**Pull request by akarabashov**

### CODE REVIEW

File: docs/design/noc-root-cert-design.md [hunk 1]
```diff
@@ -15,48 +18,64 @@ To distinguesh NOC root certificates from others, an `isNOC` boolean field will
 ## Transactions
 
 ### 1. ADD_NOC_X509_ROOT_CERTIFICATE
-This transaction adds a NOC Root Certificate owned by the Vendor.
+This transaction adds a NOC root certificate owned by the Vendor.
 
 - Who can send: Vendor account
-  - VID-scoped NOC Root Certificate: The `vid` field in the certificate's subject must be equal to the Vendor account's VID.
 - Validation:
   - The provided certificate must be a root certificate:
```
Comment (ashcherbakov) [inline docs/design/noc-root-cert-design.md:25]: Please mention, that the certificate must have a VID field set (https://github.com/zigbee-alliance/distributed-compliance-ledger/blob/master/proto/pki/certificate.proto#L26), and must be equal to the Vendor Account's VID.

File: docs/design/noc-root-cert-design.md [hunk 2]
```diff
@@ -78,24 +97,9 @@ Retrieve a list of all of NOC Root Certificates
...
\ No newline at end of file
+- Should a vendor be able to add multiple NOC root certificates with the same Subject and Subject Key Identifier combinations? If so, the vendor may want to remove a specific certificate from the list of certificates with the same Subject and Subject Key Identifier combinations.
+- How should NOC root certificate be renewed with a new one?
+- Should the `REMOVE_NOC_X509_ROOT_CERTIFICATE` operation also delete revoked certificates?
+- Should a user be able to retrieve all revoked NOC root certificates using the `GET_ALL_REVOKED_X509_NOC_ROOT_CERTS` operation?
+- In the `Joint Fabric Proposal` document, the concept of a `Trust Quotient (TQ)` is introduced as a future consideration. This concept requires adding `Add Trust` and `Revoke Trust` requests for NOCs in the DSL. Should the implementation of these requests be included in the scope of the current task?
```
Comment (ashcherbakov) [inline docs/design/noc-root-cert-design.md:105]:
```suggestion
- In the `Joint Fabric Proposal` document, the concept of a `Trust Quotient (TQ)` is introduced as a future consideration. This concept requires adding `Add Trust` and `Revoke Trust` requests for NOCs in the DCL. Should the implementation of these requests be included in the scope of the current task?
```

File: docs/design/noc-root-cert-design.md [hunk 3]
```diff
@@ -15,48 +18,64 @@ To distinguesh NOC root certificates from others, an `isNOC` boolean field will
...
-This transaction adds a NOC Root Certificate owned by the Vendor.
+This transaction adds a NOC root certificate owned by the Vendor.
 
 - Who can send: Vendor account
-  - VID-scoped NOC Root Certificate: The `vid` field in the certificate's subject must be equal to the Vendor account's VID.
 - Validation:
   - The provided certificate must be a root certificate:
     - `Issuer` == `Subject`
     - `Authority Key Identifier` == `Subject Key Identifier`
   - No existing certificate with the same `<Certificate's Issuer>:<Certificate's Serial Number>` combination.
```
Comment (ashcherbakov) [inline docs/design/noc-root-cert-design.md:28]: We need to clarify what of a certificate with the combination Subject:SKID is revoked. Should we be able to re-add it.

File: docs/design/noc-root-cert-design.md [hunk 4]
```diff
@@ -15,48 +18,64 @@ To distinguesh NOC root certificates from others, an `isNOC` boolean field will
 ## Transactions
 
 ### 1. ADD_NOC_X509_ROOT_CERTIFICATE
-This transaction adds a NOC Root Certificate owned by the Vendor.
+This transaction adds a NOC root certificate owned by the Vendor.
```
Comment (ashcherbakov) [inline docs/design/noc-root-cert-design.md:22]: Please mention, that if a certificate was removed, it should be possible to re-add it.

File: docs/design/noc-root-cert-design.md (hunk 1 above)
Comment (akarabashov) [reply docs/design/noc-root-cert-design.md:25]: As I understand, the provided PEM certificate does not include a VID field. Please correct me if I am mistaken

Comment (ashcherbakov) [reply docs/design/noc-root-cert-design.md:25]: Yes, but we have an optional VID field in certificate.proto (for VID scoped certs we require that a VID in PEM is equal to the VID field in the certificate).
So, the question is should we require a VID field to be a must-have for NOC root transaction. I think yes, and if so, the provided VID value must be equal to the Account's VID.

Comment (akarabashov) [reply docs/design/noc-root-cert-design.md:26]: Corrected the transaction description and added a question regarding the necessity of including the VID parameter in this transaction.

File: docs/design/noc-root-cert-design.md (hunk 2 above)
Comment (akarabashov) [reply docs/design/noc-root-cert-design.md]: Fixed

File: docs/design/noc-root-cert-design.md (hunk 3 above)
Comment (akarabashov) [reply docs/design/noc-root-cert-design.md:29]: Mentioned in the REVOKE_NOC_X509_ROOT_CERTIFICATE transaction section

File: docs/design/noc-root-cert-design.md (hunk 4 above)
Comment (akarabashov) [reply docs/design/noc-root-cert-design.md:22]: Mentioned in the REMOVE_NOC_X509_ROOT_CERTIFICATE transaction section

**Review by ashcherbakov — APPROVED** [review]

**Pull request merged**